src/base/static/components/molecules/user-menu.js

```diff
@@ -0,0 +1,126 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { translate } from "react-i18next";
+import { Link } from "../atoms/typography";
+import classNames from "classnames";
+import LegacyUtil from "../../js/utils.js";
+
+class UserMenu extends React.Component {
+  state = {
+    isMenuOpen: false,
+  };
+
+  toggleMenu = () => {
+    this.setState(prevState => {
+      return {
+        ...prevState,
+        isMenuOpen: !prevState.isMenuOpen,
+      };
+    });
+  };
+
+  render() {
+    const {
+      slug: datasetSlug,
+      owner: datasetOwner,
+    } = this.props.datasetDownloadConfig;
+    const isAdmin = LegacyUtil.getAdminStatus(datasetSlug);
+    if (this.props.currentUser) {
+      // If user is logged in
+      return (
+        <nav className="user-menu authed" role="article">
+          <a href="#" id="sign-in-btn" onClick={this.toggleMenu}>
+            <img
+              className="avatar header-avatar"
+              src={this.props.currentUser.avatar_url}
+            />
+          </a>
+          <ul
+            className={classNames("menu", "sign-in-menu", {
+              "is-exposed": this.state.isMenuOpen,
+            })}
+          >
+            {isAdmin && (
+              <li style={{ paddingBottom: "16px" }}>
```

Comment:
I'm not too excited about the inline styles here and the confusing state of our CSS.

After this PR is merged, I think I'll take a pass on this component using CSS-in-JS and see if it cleans things up.

Reply:
Do we want to move the CSS used here into a `user-menu.scss` file and apply BEM styles? That's how we've handled other components ports so far.